compiler/il/Aliases.cpp

```diff
@@ -797,6 +797,10 @@ OMR::SymbolReference::setSharedShadowAliases(TR_BitVector * aliases, TR::SymbolR
       while (symRef = i.getNext())
          if (symRef->getSymbol() == self()->getSymbol())
             aliases->set(symRef->getReferenceNumber());
+
+      // include symbol reference's own shared alias bitvector
```

Comment:
Just for the record, I have tested this patch with some un-pushed changes and can confirm that it works as expected for JitBuilder unions.